### ISSUE TITLE
Compute loss on validation

### ISSUE THREAD

**Issue by SandyHelly** [rwightman/efficientdet-pytorch#30]
Hi!
Thx for this amazing work! 
I'm not very experienced yet, can you tell me how can i compute loss (only) on validation? 
Set torch.no_grad() enough for this? Cause model.eval() changes output of DetBenchTrain.

```
if not self.training:
            # if eval mode, output detections for evaluation
```

I'm little confused.

**Comment by rwightman** [comment]
No grad has no effect on that  Remove the block you are pointing at, or add an extra check like 'if not self.training and not self.loss_only_on_eval'

**Comment by SandyHelly** [comment]
Thanks for the answer! Already remove it. Just wanted to make sure I'm not missing something :)

**Comment by SandyHelly** [comment]
Thanks for the answer! Already remove it. Just wanted to make sure I'm not missing something :)
